pkg/rancher-desktop/main/extensions/extensions.ts

```diff
@@ -103,6 +104,26 @@ export class ExtensionImpl implements Extension {
     return this._metadata as Promise<ExtensionMetadata>;
   }
 
+  /** Extension image labels */
+  get labels(): Promise<Record<string, string>> {
+    this._labels ??= (async() => {
+      if (await this.isInstalled()) {
+        const labelPath = path.join(this.dir, 'labels.json');
+
+        return JSON.parse(await fs.promises.readFile(labelPath, 'utf-8'));
+      }
+
+      const info = await this.client.runClient(
+        ['image', 'inspect', '--format={{ json .Config.Labels }}', this.image],
+        'pipe',
+        { namespace: ExtensionImpl.extensionNamespace });
+
+      return JSON.parse(info.stdout);
+    })();
+
+    return this._labels as Promise<Record<string, string>>;
```

Comment:
Should we be wrapping any of this in a `try...catch` statement to handle any errors involved with reading the file or parsing `stdout`?